Applications/FileIO/SWMM/SWMMInterface.h

```diff
@@ -0,0 +1,245 @@
+    bool convertSwmmInputToLineMesh();
+
+    /// Reading points from SWMM input file and converting them into OGS point-type vector.
+    /// This method is shared by geometry- and mesh conversion.
+    template <typename T>
+    static bool readCoordinates(std::ifstream &in, std::vector<T*> &points, std::vector<std::string> &names)
+    {
+        bool finished (false);
+        std::size_t id (points.size());
+        std::string line;
+        getline(in, line);
+        while (!isSectionFinished(line))
+        {
+            if (isCommentLine(line))
+            {
+                getline(in, line);
+                continue;
+            }
+
+            std::vector<std::string> split_str (BaseLib::splitString(line));
+            if (split_str.size() != 3)
+            {
+                ERR ("Format not recognised.");
+                return false;
+            }
+            names.push_back(split_str[0]);
+
+            double const x = BaseLib::str2number<double>(split_str[1]);
+            double const y = BaseLib::str2number<double>(split_str[2]);
+            T* pnt = new T(x, y, 0, id);
+            points.push_back(pnt);
+            id++;
+            getline(in, line);
+        }
+        return true;
+    }
+
+    /// Reads input information associated with nodes (elevation, depth, etc.)
+    bool readNodeData(std::ifstream &in, std::vector<MeshLib::Node*> &nodes,
+        std::map<std::string, std::size_t> const& name_id_map,
+        std::vector<double> &max_depth, bool read_max_depth);
+
+    /// Reads links/conduits and returns them as a vector of OGS line elements
+    bool readLineElements(std::ifstream &in, std::vector<MeshLib::Element*> &elements,
+        std::vector<MeshLib::Node*> const& nodes, std::map<std::string, std::size_t> const& name_id_map);
+
+    /// Reads subcatchment information
+    bool readSubcatchments(std::ifstream &in, std::map< std::string, std::size_t> const& name_id_map);
+
+    /// Reads pollutant names and parameters
+    bool readPollutants(std::ifstream &in);
+
+    /// Returns the name of an array (or an empty string if errors occured)
+    std::string getArrayName(SwmmObject obj_type, std::size_t var_idx, std::size_t n_pollutants) const;
+
+    /// Reads the location of external rain gauge time series files.
+    bool addRainGaugeTimeSeriesLocations(std::ifstream &in);
+
+    /// Matches existing subcatchment names with subsequently read polylines marking the outlines of said subcatchments.
+    bool matchSubcatchmentsWithOutlines(std::vector<GeoLib::Polyline*> const& lines, std::vector<std::string> const& names);
+
+    /// Creates a temporary string vector containing all subcatchment names in correct order
+    std::vector<std::string> getSubcatchmentNameMap() const;
+
+    /// During geometry conversion, this adds elevation values to the existing point vector
+    static bool SwmmInterface::addPointElevation(std::ifstream &in,
```

Comment:
This converts points (x, y, 0) -> (x, y, z)? Is that not yet implemented somewhere in OGS?